lib/node_modules/@stdlib/repl/lib/filter_by_prefix.js

```diff
@@ -32,13 +33,29 @@ var startsWith = require( '@stdlib/string/starts-with' );
 * @param {Array} out - output array
 * @param {Array} arr - source array
 * @param {string} str - string filter
+* @param {boolean} isFuzzy - boolean indicating if the completions should be fuzzy
 * @returns {Array} output array
 */
-function filterByPrefix( out, arr, str ) {
+function filterByPrefix( out, arr, str, isFuzzy ) {
+	var fuzzyResults = [];
+	var match;
 	var i;
-	for ( i = 0; i < arr.length; i++ ) {
-		if ( startsWith( arr[ i ], str ) ) {
-			out.push( arr[ i ] );
+
+	if ( isFuzzy ) {
+		for ( i = 0; i < arr.length; i++ ) {
+			match = fuzzyMatch( arr[ i ], str );
+			if ( match ) {
+				fuzzyResults.push( match );
```

Comment:
Why do we need `fuzzyResults` here. Can we not directly push to the `out` array instead?

Reply:
You're right, don't know how I got there...